src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/GoogleDataprocDAO.scala

```diff
@@ -28,17 +30,16 @@ import com.google.api.services.storage.{Storage, StorageScopes}
 import org.broadinstitute.dsde.workbench.google.GoogleUtilities
 import org.broadinstitute.dsde.workbench.leonardo.config.{DataprocConfig, ProxyConfig}
 import org.broadinstitute.dsde.workbench.leonardo.model.ClusterStatus.{ClusterStatus => LeoClusterStatus}
-import org.broadinstitute.dsde.workbench.leonardo.model.ModelTypes.GoogleProject
-import org.broadinstitute.dsde.workbench.leonardo.model.{ClusterErrorDetails, ClusterRequest, ClusterResponse, GoogleBucketUri, LeoException, StorageObjectResponse, Cluster => LeoCluster, ClusterStatus => LeoClusterStatus}
+import org.broadinstitute.dsde.workbench.leonardo.model.{BucketObject, ClusterErrorDetails, ClusterName, ClusterRequest, ClusterResponse, FirewallRuleName, GoogleBucket, GoogleBucketUri, GoogleProject, IP, InstanceName, LeoException, OperationName, StorageObjectResponse, Zone, Cluster => LeoCluster, ClusterStatus => LeoClusterStatus}
 
 import scala.collection.JavaConverters._
 import scala.concurrent.{ExecutionContext, Future, blocking}
 
-case class CallToGoogleApiFailedException(googleProject: GoogleProject, clusterName:String, exceptionStatusCode: Int, errorMessage: String)
-  extends LeoException(s"Call to Google API failed for $googleProject/$clusterName. Message: $errorMessage",exceptionStatusCode)
+case class CallToGoogleApiFailedException(googleProject: GoogleProject, context: String, exceptionStatusCode: Int, errorMessage: String)
```

Comment:
There are many Google API Exceptions which do not involve a specific cluster

Reply:
Should the different types be different exceptions? Like a cluster-y exception and an operation-y exception etc. Not sure, just wondering.

Reply:
I did that in a previous iteration and switched to this after I realized the only difference was a string describing the context.